cli/internal/client/client.go

```diff
@@ -17,9 +18,10 @@ import (
 
 type ApiClient struct {
 	// The api's base URL
-	baseUrl      string
-	Token        string
-	turboVersion string
+	baseUrl          string
+	Token            string
+	turboVersion     string
+	CurrentFailCount uint64
```

Comment:
As best I can tell, we are not checking this value before initiating a new request on an existing `ApiClient`, only for retries after each request fails. Is that intended?

Reply:
Ah, my mistake, this happens outside of the `ApiClient`. It might be a good idea to encapsulate that a bit better, so that calling code does not need to check.